### ISSUE TITLE
Latest version makes deleting local branches a pro feature

### ISSUE THREAD

**Issue by b-johnse** [gitkraken/vscode-gitlens#3514]
### Description

Right click on a local branch > delete > asks to go pro instead of just deleting the branch.

This is not a pro feature, its a shortcut for a cli command

### GitLens Version

15.4.0

### VS Code Version

Version: 1.92.2 (user setup)
Commit: fee1edb8d6d72a0ddff41e5f71a671c23ed924b9
Date: 2024-08-14T17:29:30.058Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by jkelroy** [comment]
Hello @b-johnse, I am not seeing the same behavior. Where are you deleting the branch from? Are you not logged in or are you logged in with a free account?

Can you please enable debug logging by running from _GitLens: Enable Debug Logging_ from the Command Palette (<kbd>F1</kbd> or <kbd>ctrl/cmd</kbd>+<kbd>shift</kbd>+<kbd>p</kbd>). This will enable logging to the _GitLens_ and _GitLens (Git)_ channel in the _Output_ pane.

Once enabled, please reproduce the issue, and attach the log from the both _GitLens_ and _GitLens (Git)_ channels.

**Comment by eamodio** [comment]
@b-johnse are you trying to delete a branch that is on a worktree? If so, that could be why you are seeing this behavior, as worktrees are a Pro feature. Prior to the 15.4 release, GitLens wouldn't properly detect that the branch was on a worktree and let you think you could delete it, but then it would fail, because it is checked out on a worktree. Now in 15.4 we properly detect that can offer to delete the worktree first and then the branch, but since worktrees are Pro, you would see a that gated.

**Comment by eamodio** [comment]
I think we tracked down the case you might have been running into -- there was an issue where if you had VS Code open to a worktree, but tried to delete a local branch checked out on the "main" worktree we would incorrectly even allow that case (git won't let you delete the checked out branch on the main worktree) but it would also end up triggering our new worktree detection (which also shouldn't happen for the "main" worktree).

Can you please verify this fix in tomorrow's (`v2024.09.05xx`) pre-release edition of GitLens? Thank you so much!

You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/228608740-6071bc19-7b1a-44c8-843f-d2af44d6114b.png)

**Comment by b-johnse** [comment]
@eamodio yes the pre-release version it work again! 
I don't do anything with worktrees and my settings have ` "gitlens.plusFeatures.enabled": false`  so sounds like you correctly found it

**Comment by nzaytsev** [comment]
@b-johnse - this is a quite strange case. The `upgrade to pro` prompt should not appear in case that you described. We would like to fully understand what has happened. Can we ask you to try reproduce it again (switch back to release version from the extension page) and make a few additional things for us to understand.

![Screenshot 2024-09-06 at 08 54 49](https://github.com/user-attachments/assets/db674f9f-4e89-4b20-9145-1aaa2c8db51f)

Currently, we'd like to see 2 things:
1. Could you record the video of the reproduced issue?
2. We need to be sure if you really don't use worktrees. Execute please the below command in your terminal in the project directory and put the result here with a comment
```sh
git worktree list
```
![Screenshot 2024-09-06 at 08 55 33](https://github.com/user-attachments/assets/d477e72b-1b3e-468b-8f66-c5091a033a92)

**Comment by dmamchyts** [comment]
The same issue.

Steps for reproducing:

1) Remove current gitlens extension
2) Install from vscode extension: GitLens — Git supercharged
3) profit:

![image](https://github.com/user-attachments/assets/ad33c21f-ec2a-4762-b527-f1f7f75f805c)

**Comment by nzaytsev** [comment]
@dmamchyts - thank you for your comment. Could you describe all the steps you made to reproduce it? See my [previous comment](https://github.com/gitkraken/vscode-gitlens/issues/3514#issuecomment-2333037386) with the info that we need. Also, know that we expect this bug to be fixed in pre-release version, you can check it according to [the comment](https://github.com/gitkraken/vscode-gitlens/issues/3514#issuecomment-2330575954) of @eamodio

**Comment by dmamchyts** [comment]
@nzaytsev  thank you. After installing pre-release version all works fine.

**Comment by sittingpilgrim** [comment]
I'm getting the error too - haven't yet tried updating.  Not using worktrees.  My basic workflow re branch creation/destruction goes like this:

- create branch off of main, usually using Jira to create the branch from the associated ticket
- create a local, tracking branch in VSCode using bash: git checkout -t origin/branch-name
- work, commit, create PR, and then merge.  For the most recent case the merge was done via the 'Squash and Merge' on the github PR site.  Sometimes I use the github app on my phone.  Sometimes I create the PR via VSCode, sometimes via Jira/github.
- switch to main branch in VSCode/gitlens
- right-click on tracking branch and choose Delete Branch - it now prompts as [shown earlier](https://github.com/gitkraken/vscode-gitlens/issues/3514#issuecomment-2338408844)

Switched to pre-release version and it works as it has in the past.

Thanks for your work on this extension.

**Comment by DmitryEfimenko** [comment]
Confirming the issue.
My case:
1. created a feature branch of main.
2. Pushed it to Github.
3. Merged to main in Github.
4. Attempting to delete feature branch in VsCode asks for Pro access.

Tried switching to Pre-release version - didn't help.

**Comment by nzaytsev** [comment]
Well, as I can not reproduce it, let's try other way. I'll describe all steps I do and you @DmitryEfimenko, @sittingpilgrim, @b-johnse will point to the place where I'm wrong. 

0. check environment

- I use the released GitLens version 15.4.0
- I have a free license and private repo
![Screenshot 2024-09-11 at 13 03 47](https://github.com/user-attachments/assets/8dc6f2d4-4ead-4b55-98ef-2efe2d8ab798)

1. Create a branch from remote. For example from github issue page
     - create a new issue on GitHub using the `new issue` button 
![Screenshot 2024-09-11 at 12 39 51](https://github.com/user-attachments/assets/2acdcfb7-0498-4a59-a336-b65358566b13)
      - add a dev branch to the issue using the button from development block 
![Screenshot 2024-09-11 at 12 40 33](https://github.com/user-attachments/assets/8a2aa2c9-333c-4d02-b2fc-6e765775959c)
![Screenshot 2024-09-11 at 12 41 20](https://github.com/user-attachments/assets/7b2f2f7f-76e9-4f96-9157-e6250ec13f31)

2. Checkout locally the branch
    - open VSCode with current repo folder
    - fetch repo

```sh
git fetch
```

    - checkout with --track flag

```sh
git checkout -t origin/53-pro-message-test-issue 
```

![Screenshot 2024-09-11 at 12 45 01](https://github.com/user-attachments/assets/e0c64ead-f8eb-4855-acfc-705aad662c49)

2.1 Let's check if the branch is in current worktree:
   - run `git worktree list` and see that the branch is the first in the list
![Screenshot 2024-09-11 at 12 50 41](https://github.com/user-attachments/assets/55d51c66-e2aa-4445-8157-8987ab855801)



3. Now do some work

![Screenshot 2024-09-11 at 12 47 45](https://github.com/user-attachments/assets/43cdf6e8-c752-41e0-8c34-b4f34cd3264c)

4. Commit it. I use the IDE UI
    - add some message. for me it's `pro message test 3`
![Screenshot 2024-09-11 at 12 49 24](https://github.com/user-attachments/assets/ea0f780b-f46e-42cf-b100-e182fd75f61e)
    - click Commit
    - push changes
![Screenshot 2024-09-11 at 12 51 52](https://github.com/user-attachments/assets/4ac5ed47-2973-4a3f-85f6-c11f23343779)

5. Well, let's create a pull request and merge in to the main
   - go to the repo github
   - see the notification about recently pushed commit
![Screenshot 2024-09-11 at 12 52 53](https://github.com/user-attachments/assets/57f73b64-dc02-4f96-a77b-c58794f8fdef)
   - just create a pull request 
![Screenshot 2024-09-11 at 12 53 36](https://github.com/user-attachments/assets/966ea05a-674b-4dcf-b867-624461da42ed)
   - merge the pr from github
![Screenshot 2024-09-11 at 12 54 39](https://github.com/user-attachments/assets/3f46895e-e474-482c-9f0c-3783566b245b)
   - see that the branch is not deleted
![Screenshot 2024-09-11 at 12 54 56](https://github.com/user-attachments/assets/25dff8a3-db44-4ca8-8267-379cdd61e8e8)

6. Try to reproduce the branch deletion issue
   - go to the VSCode window
   - checkout to main

```sh
git checkout main
```

![Screenshot 2024-09-11 at 12 56 19](https://github.com/user-attachments/assets/cad5b1d2-3f85-43ae-8391-547f12755d4a)

   - try to delete the branch `53-pro-message-test-issue` from the branches view

![Screenshot 2024-09-11 at 12 56 59](https://github.com/user-attachments/assets/a377bf79-bd1b-44d4-b7fd-2249dffd1b44)

   - see usual confirmation

![Screenshot 2024-09-11 at 12 57 08](https://github.com/user-attachments/assets/02cb9431-b334-4e21-8bb7-b95b2a098437)

The workflow is finished successfully

**Comment by DmitryEfimenko** [comment]
There's now potentially a related issue. The "Delete Branch" option is completely gone. Previously, the "Delete option" would not show when you are on the branch that you want to delete. But, I made sure it's not the case.

Also, see the result of these commands:
```sh
git worktree list
C:/Source/art/art-backend  b5033fb [master]
```

```sh
git branch
  fix/75
* master
```


![image](https://github.com/user-attachments/assets/6c3fe98f-f222-4c85-b4cc-8342186f41a3)

**Comment by eamodio** [comment]
@DmitryEfimenko Thanks for the extra information, I've opened (and fixed) #3582 to track this additional issue.

**Comment by jbosse** [comment]
I am using v15.6.0 and the "Delete Branch" option is visible but disabled.![Image](https://github.com/user-attachments/assets/3a759a5b-5bf5-4458-a66c-52cb74c72aed)

**Comment by eamodio** [comment]
@jbosse There is no difference between those versions (or any version in the last ~20 months since the enablement control was added) with regard to the enablement of those commands. It is just likely that the restart of VS Code fixed/resolved a stuck/long(?) Git operation.

VS Code sets an `operationInProgress` context key when it (the built-in Git extension) runs some Git operations (and clears/resets that key once the operation is complete). Both VS Code and GitLens honor/use that context key to disable certain Git operations while another is currently running and they get re-enabled once the operation completes.

I've never seen a case where `operationInProgress` isn't properly reset, but if it persists or is reproducible (outside of a normal running Git operations) I would suggest opening an issue with VS Code. As currently GitLens never modifies that context key, only reads from it.

**Comment by jbosse** [comment]
@eamodio Thank you. I did update back and it is working correctly in v15.6.0. I had just rebooted so "turning it off and on again" wasn't even in my mind. I've never had this happen until this morning so I am am willing to attribute it to cosmic rays. :-)
